src/main/java/org/testng/internal/annotations/AnnotationHelper.java

```diff
@@ -163,10 +163,12 @@ private static IConfigurationAnnotation createConfiguration(
     }
     if (bg != null) {
       result.setBeforeGroups(bg.getBeforeGroups());
+      result.setBeforeGroups(true);
```

Comment:
Code smells: 2 methods with the same name but different signature

Reply:
@juherr - I didn't have a choice. The `setBeforeGroups(boolean)` is basically being used to flag that an annotation is a `@BeforeGroups` representation. The name is already taken. For lack of a better nomenclature, I just decided to overload the method and provide another variant.